src/libraries/math/util_math.h

```diff
@@ -22,39 +22,318 @@
 
  */
 
+#include "turtle.h"
+#include "util_math.h"
+#include "constants.h"
+#include "math.optim.h"
 
-short tridiag(short a, long r, long c, long nx, Vector<double> *diag_inf,
-              Vector<double> *diag, Vector<double> *diag_sup, Vector<double> *b, Vector<double> *e);
+/*----------------------------------------------------------------------------------------------------------*/
 
-short tridiag2(short a, long r, long c, long nbeg, long nend,
+inline short tridiag(short a, long r, long c, long nx, Vector<double> *diag_inf,
```

Comment:
usually, you inline small functions, not big ones. I don't think the compiler will actually inline the tridiag fuctions. It's ok for now. Just to let you know

Reply:
My idea was to inline ```adaptiveSimpsonsAux2``` so since in util.math there were also small functions I tried to inline all of them.